genai-perf/genai_perf/parser.py

```diff
@@ -246,6 +246,49 @@ def _check_load_manager_args(args: argparse.Namespace) -> argparse.Namespace:
     return args
 
 
+def is_valid_url(url):
+    """
+    Checks if a URL is valid. It must use 'http' or 'https', have a valid
+    netloc, optional port, and contain '/metrics' in the path.
+    """
+    try:
+        result = urlparse(url)
+
+        # Check scheme
+        if result.scheme not in ["http", "https"]:
+            return False
+
+        # Check netloc
+        if not result.netloc:
+            return False
+
+        # Check path contains '/metrics'
+        if "/metrics" not in result.path:
+            return False
+
+        return True
+    except Exception:
+        return False
```

Comment:
Do we just want to return false here?

Reply:
This function returns if url is valid or not and the _check_server_metrics_url will display error message based on the return value.